src/TensorFlowNET.Core/Buffers/Buffer.cs

```diff
@@ -93,38 +90,29 @@ private static IntPtr _toBuffer(byte[] data)
             }
         }
 
-        public static implicit operator IntPtr(Buffer buffer)
-        {
-            buffer.EnsureNotDisposed();
-            return buffer._handle;
-        }
-
-        public static explicit operator byte[](Buffer buffer) => buffer.ToArray(); //has to be explicit, developer will assume it doesn't cost.
-
         /// <summary>
         ///     Copies this buffer's contents onto a <see cref="byte"/> array.
         /// </summary>
         public byte[] ToArray()
         {
-            EnsureNotDisposed();
-
-            unsafe
+            using (Handle.Lease())
             {
+                ref readonly TF_Buffer buffer = ref DangerousBuffer;
                 var len = buffer.length;
                 if (len == 0)
                     return Array.Empty<byte>();
 
-                byte[] data = new byte[len];
-                fixed (byte* dst = data)
-                    System.Buffer.MemoryCopy((void*) bufferptr->data, dst, len, len);
-
-                return data;
+                unsafe
+                {
+                    var source = new ReadOnlySpan<byte>(buffer.data.ToPointer(), checked((int)len));
+                    var data = new byte[len];
+                    source.CopyTo(data);
```

Comment:
Doesn't need `fixed` keyword anymore? Is safe for coping data which might be moved by GC.

Reply:
This is copying directly to `byte[]`, so there is no need to pin `data` to get a `byte*`.

Reply:
➡️ I went ahead and simplified this one step further.